scripts/createSingleNav.js

```diff
@@ -0,0 +1,47 @@
+const fs = require('fs');
+const path = require('path');
+const yaml = require('js-yaml');
+
+const createSingleNav = () => {
+  const rootNavYaml = yaml.safeLoad(
+    fs.readFileSync(path.join(process.cwd(), 'src/nav/root.yml'), 'utf8')
+  );
+
+  const ignorePages = [
+    '/docs/glossary/glossary',
+    '/attribute-dictionary',
+    '/whats-new',
+    '/docs/release-notes',
+  ];
+
+  const rootNavChildren = rootNavYaml.pages.reduce((acc, page) => {
+    let subNavChildren = null;
+    if (
+      page.path &&
+      !page.path.startsWith('http') &&
+      !ignorePages.includes(page.path)
+    ) {
+      const pageDirectory = page.path.split('/')[2];
+      subNavChildren = yaml.safeLoad(
+        fs.readFileSync(
+          path.join(process.cwd(), `src/nav/${pageDirectory}.yml`),
+          'utf8'
+        )
+      );
+    }
+    if (subNavChildren) {
+      acc.push({ ...page, pages: [subNavChildren] });
+    } else {
+      acc.push({ ...page });
+    }
+    return acc;
+  }, []);
+  const navToRuleThemAll = { ...rootNavYaml, pages: rootNavChildren };
+  const navToRuleThemAllYaml = yaml.safeDump(navToRuleThemAll, {
+    lineWidth: 99999,
+  });
+
+  fs.writeFileSync(`src/nav/generatedNav.yml`, navToRuleThemAllYaml);
```

Comment:
i see the build preview had a 40m build time.

do you think that's due to building this file? if so, is there another way to build this file outside the build and just source it during build? adds some complexity, but might be worth it if it saves us ~10min of build time

Reply:
i've seen up to 47 min lately with develop. the script itself is really quick so i'd be surprised! but we should keep an eye on it

Reply:
oh god. really? that's...a long time.